src/lib.rs

```diff
@@ -1207,37 +1323,38 @@ impl Node {
 		};
 
 		let payment_hash = PaymentHash((*invoice.payment_hash()).into_inner());
-		inbound_payments_lock.insert(
+		let payment_info = PaymentInfo {
+			preimage: None,
 			payment_hash,
-			PaymentInfo {
-				preimage: None,
-				secret: Some(*invoice.payment_secret()),
-				status: PaymentStatus::Pending,
-				amount_msat,
-			},
-		);
+			secret: Some(invoice.payment_secret().clone()),
+			amount_msat,
+			direction: PaymentDirection::Inbound,
+			status: PaymentStatus::Pending,
+		};
+
+		self.payment_store.insert(payment_info)?;
+
 		Ok(invoice)
 	}
 
 	/// Query for information about the status of a specific payment.
-	pub fn payment_info(&self, payment_hash: &[u8; 32]) -> Option<PaymentInfo> {
-		let payment_hash = PaymentHash(*payment_hash);
-
-		{
-			let outbound_payments_lock = self.outbound_payments.lock().unwrap();
-			if let Some(payment_info) = outbound_payments_lock.get(&payment_hash) {
-				return Some((*payment_info).clone());
-			}
-		}
-
-		{
-			let inbound_payments_lock = self.inbound_payments.lock().unwrap();
-			if let Some(payment_info) = inbound_payments_lock.get(&payment_hash) {
-				return Some((*payment_info).clone());
-			}
-		}
+	pub fn payment_info(&self, payment_hash: &PaymentHash) -> Option<PaymentInfo> {
+		self.payment_store.get(payment_hash)
+	}
 
-		None
+	/// List node's connected peers.
+	pub fn list_peers(&self) -> Vec<PeerDetails> {
+		let active_connected_peers: Vec<PublicKey> =
+			self.peer_manager.get_peer_node_ids().iter().map(|p| p.0).collect();
+		self.peer_store
```

Comment:
yeah... I just took a stab at it and it looks like it may be easier if we both had `PeerInfo` and the tuple returns by `get_peer_node_ids` return a consistent type for address.